src/doc.rs

```diff
@@ -0,0 +1,354 @@
+    let mut module_doc = HashMap::new();
+
+    let mut docs = Vec::<EEP48Doc>::new();
+    for statement in module.statements.iter() {
+      match statement {
+        Statement::Fn {
+          meta,
+          name,
+          args,
+          public,
+          return_annotation,
+          ..
+        } => {
+          let public_str = if *public { "pub " } else { "" };
+          let mut parsed_arg_vec: Vec<String> = vec![];
+          for arg in args {
+            if let Some(a) = arg_to_string(arg) {
+              parsed_arg_vec.push(a)
+            };
+          }
+          let parsed_args = parsed_arg_vec.join(", ");
+          let return_type = return_annotation.as_ref().map(|a| type_to_string(&a));
+          let fn_docs: Option<HashMap<String, String>> = self
+            .find_block_for_line(meta.start)
+            .map(|d| vec![("en-US".to_string(), d)].into_iter().collect());
+          docs.push(EEP48Doc {
+            name: name.to_string(),
+            arity: args.len(),
+            signature: vec![match return_type {
+              Some(retrn) => format!("{} fn {}({}) -> {}", public_str, name, parsed_args, retrn),
+              None => format!("{} fn {}({})", public_str, name, parsed_args),
+            }],
+            doc: fn_docs,
+            meta: DocMeta::new("".to_string()),
+            typ: DocType::Fn,
+          });
+        }
+        Statement::TypeAlias {
+          meta,
+          alias,
+          args,
+          resolved_type,
+          public,
+          comments,
+        } => {
+          ();
+        }
+        Statement::CustomType {
+          meta,
+          name,
+          args,
+          public,
+          constructors,
+          comments,
+        } => {
+          ();
+        }
+        Statement::ExternalFn {
+          meta,
+          public,
+          args,
+          name,
+          retrn,
+          module,
+          fun,
+          comments,
+        } => {
+          ();
+        }
+        Statement::ExternalType {
+          meta,
+          public,
+          name,
+          args,
+          comments,
+        } => {
+          ();
+        }
+
+        Statement::Import { .. } => {
+          ();
+        }
+      }
+    }
+    // TODO: Not the real logic
+    module_doc.insert(
+      "en-US".to_string(),
+      self
+        .blocks
+        .first()
+        .unwrap_or(&vec![])
+        .iter()
+        .map(|rc| rc.content.to_string())
+        .join("\n")
+        .to_string(),
+    );
+    EEP48DocChunk {
+      anno: ErlAnno {
+        line: 0,
+        column: 0,
+        file: module.name_string(),
+      },
+      module_doc: module_doc,
+      meta: DocMeta {
+        authors: None,
+        deprecated: None,
+        since: None,
+        cross_references: None,
+        edit_url: format!("https://www.example.com/{}", module.name_string()),
+      },
+      docs,
+    }
+  }
+}
+
+fn type_to_string(typ: &TypeAst) -> String {
+  "".to_string()
```

Comment:
We can use the `typ::pretty` module for this :)

Reply:
Awesome.  I'm glad I don't have to reinvent the wheel.